google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/MavenBuildDeploymentSourceType.java

```diff
@@ -107,6 +110,11 @@ public void save(@NotNull ModuleDeploymentSource deploymentSource, @NotNull Elem
       if (deployable.getVersion() != null) {
         tag.setAttribute(VERSION_ATTRIBUTE, ((AppEngineDeployable) deploymentSource).getVersion());
       }
+
+      if (deployable.getEnvironment() != null) {
```

Comment:
Well spotted. It is possible for tag.getAttributeValue() to return null, so I'll guard valueOf above from nulls, in case this triggers for existing Maven artifacts with artifact set to null.